src/lib/holocene/markdown-editor/markdown-editor.svelte

```diff
@@ -0,0 +1,80 @@
+<script lang="ts">
+  import { markdown } from '@codemirror/lang-markdown';
+  import { EditorState } from '@codemirror/state';
+  import { EditorView, lineNumbers } from '@codemirror/view';
+  import { onMount, tick } from 'svelte';
+
+  import TabButton from '$lib/holocene/tab-buttons/tab-button.svelte';
+  import TabButtons from '$lib/holocene/tab-buttons/tab-buttons.svelte';
+  import { useDarkMode } from '$lib/utilities/dark-mode';
+  import { getEditorThemeWithLineNumbers } from '$lib/vendor/codemirror/custom-extensions';
+
+  import MarkdownPreview from './markdown-preview.svelte';
+
+  interface Props {
+    content: string;
+  }
+
+  let { content = $bindable('') }: Props = $props();
+  let editorElement: HTMLElement | null = $state(null);
+  let editorView: EditorView | null = $state(null);
+  let activeTab: 'edit' | 'preview' = $state('edit');
+
+  const createEditorView = () => {
+    return new EditorView({
+      parent: editorElement,
+      state: EditorState.create({
+        doc: content,
+        extensions: [
+          markdown(),
+          lineNumbers(),
+          getEditorThemeWithLineNumbers($useDarkMode, false),
+        ],
+      }),
+      dispatch(transaction) {
+        editorView.update([transaction]);
+        if (transaction.docChanged) {
+          content = editorView.state.doc.toString();
+        }
+      },
```

Comment:
Lol I got to here and it seems less feasible but I'm curious when this function ends up getting called in the lifecycle of codemirror. 

Probably ignore the comments above.